### PR TITLE
DFP commercial line items should be validated against active adunits

### PR DESCRIPTION
**Pull request by rich-nguyen**
This fixes an issue where line items targeting inactive/archived ad units were being rejected from our system. 

The correct behaviour is to reject unknown adunits, instead of rejecting adunits which are not active.

Note to self: This will invalidate the format for: line items report, high merch report, and paid for tags report. In the long term, its better to do this invalidation than use an `Option` for something that will always be there.

### CODE REVIEW

File: admin/app/dfp/AdUnitAgent.scala [hunk 1]
```diff
@@ -41,8 +38,27 @@ object AdUnitAgent extends DataAgent[String, GuAdUnit] {
 
 object AdUnitService {
 
+  // Retrieves the ad unit object if the id matches and the ad unit is active.
```
Comment (JonNorman) [inline admin/app/dfp/AdUnitAgent.scala]: Would it be worth calling this `activeAdUnit` just to be explicit (and complete with `archivedAdUnit` to make it clear that they are non-intersecting sets)?

File: admin/app/dfp/DataValidation.scala [hunk 2]
```diff
@@ -16,7 +16,12 @@ object DataValidation {
         toSeq(inventoryTargeting.getTargetedAdUnits).map(_.getAdUnitId()
       )).getOrElse(Nil)
 
-    dfpAdUnitIds.forall( adUnitId => {
+    // The validation should not account for inactive or archived ad units.
+    val activeDfpAdUnits = dfpAdUnitIds.filterNot { adUnit =>
```
Comment (JonNorman) [inline admin/app/dfp/DataValidation.scala]: Perhaps change the variant name to `adunitId` to be consistent with that below

**Comment by JonNorman** [comment]
Apart from the tiny comments, looks good to me :+1:

File: admin/app/dfp/AdUnitAgent.scala (hunk 1 above)
Comment (rich-nguyen) [inline admin/app/dfp/AdUnitAgent.scala]: yeah good idea

**Pull request merged**